src/open62541_compat.cpp

```diff
@@ -269,6 +268,8 @@ OpcUaType UaVariant::type() const
         return OpcUaType_String;
     else if(m_impl->type == &UA_TYPES[UA_TYPES_BOOLEAN])
     	return OpcUaType_Boolean;
+    else if(m_impl->type == &UA_TYPES[UA_TYPES_BYTESTRING])
+    	return OpcUaType_ByteString;
```

Comment:
Hi Ben, I looked UA_TYPES up, in fact it is open62541's type dictionary. A short fragment:
/* Boolean */
{ .typeId = {.namespaceIndex = 0, .identifierType = UA_NODEIDTYPE_NUMERIC, .identifier.numeric = 1},
  .typeIndex = UA_TYPES_BOOLEAN,
#ifdef UA_ENABLE_TYPENAMES
  .typeName = "Boolean",
#endif
  .memSize = sizeof(UA_Boolean),
  .builtin = true,
  .fixedSize = true,
  .overlayable = true,
  .membersSize = 1,
  .members = Boolean_members },

so, good news: we   have the numeric identifier there. I'll merge UaByteString support as it is since I didn't see any other objection and create a new improvement to fix this.
OK?